docs/examples/grid_parameters.py

```diff
@@ -0,0 +1,298 @@
+chi2_F0 = pint.gridutils.grid_chisq(f, ("F0",), (F0,))
+
+# %%
+# find the uncertainty by looking where Delta chi^2=1
+F0_delta_chi2 = np.interp(
+    1,
+    chi2_F0[F0 >= f.model.F0.quantity] - chi2_F0.min(),
+    F0[F0 >= f.model.F0.quantity].astype(np.float64),
+)
+
+# %%
+fig, ax = plt.subplots(figsize=(16, 9))
+# just plot the values offset from the best-fit values
+ax.plot(F0 - f.model.F0.quantity, chi2_F0, label="$\chi^2$ curve")
+ax.plot(0, chi2_F0.min(), "ro", markersize=12, label="Minimum $\chi^2$")
+ax.plot((f.model.F0.uncertainty) * np.ones(2), [59, 69], "r:", label="Uncertainty")
+ax.plot((-f.model.F0.uncertainty) * np.ones(2), [59, 69], "r:")
+ax.plot(
+    [
+        F0_delta_chi2 - f.model.F0.quantity,
+        F0_delta_chi2 - f.model.F0.quantity,
+        -(F0_delta_chi2 - f.model.F0.quantity),
+        -(F0_delta_chi2 - f.model.F0.quantity),
+    ],
+    [chi2_F0.min(), chi2_F0.min() + 1, chi2_F0.min() + 1, chi2_F0.min()],
+    "g--",
+    label="$\Delta \chi^2=1$",
+)
+# ax.errorbar(0,0,xerr=f.model.F0.uncertainty,yerr=f.model.F1.uncertainty,fmt='ro')
+ax.set_xlabel("$\Delta F_0$ (Hz)", fontsize=24)
+ax.set_ylabel("$\chi^2$", fontsize=24)
+ax.set_ylim([59, 69])
+ax.legend()
+
+# %% [markdown]
+# From the above you can see that the calculated uncertainties on $F0$ from the model fit agree with where $\Delta \chi^2=1$, which is what we would expect for a single parameter.
+
+# %% [markdown]
+# # 2D Grid
+
+# %% [markdown]
+# Now, compute a 2D grid of $\chi^2(F0,F1)$
+
+# %% tags=[]
+chi2grid = pint.gridutils.grid_chisq(f, ("F0", "F1"), (F0, F1))
+
+# %% [markdown]
+# We want to plot contour levels appropriate for the joint confidence contours with 2 parameters.  This is discussed many places (e.g., https://ned.ipac.caltech.edu/level5/Wall2/Wal3_4.html, or Chapter 15.6 of Numerical Recipes in C - in particular look at Figure 15.6.4).
+# Rather than look up the values in a table we will compute them ourselves.
+#
+# The goal is to find $x$ such that the CDF of the $\chi^2_\nu$ (for $\nu=2$ parameters) distribution evaluated at $x$ is equal to the desired confidence interval.
+#
+# We want 1, 2, and 3 $\sigma$ confidence intervals.  So first we determine what the CDFs of the normal distribution are at those values.
+#
+# We then get a $\chi^2_2$ random variable, and interpolate to find the values of $x$ (which are our desired confidence intervals).
+#
+# We can check these against the values in the linked table (or elsewhere).
+
+# %% tags=[]
+# 1, 2, and 3 sigma confidence limits
+nsigma = np.arange(1, 4)
+# these are the CDFs going from -infinity to nsigma.  So subtract away 0.5 and double for the 2-sided values
+CIs = (scipy.stats.norm().cdf(nsigma) - 0.5) * 2
+print(f"Confidence intervals for {nsigma} sigma: {CIs}")
+# chi^2 random variable for 2 parameters
+rv = scipy.stats.chi2(2)
+x = np.linspace(0, 20)
+contour_levels = np.interp(CIs, rv.cdf(x), x)
```

Comment:
Why interpolate? Doesn't a chi-squared random variable support inverse CDF?

Reply:
Yes, it does.  I couldn't find it initially (I think it used to be icdf) but I found it now.